go/tasks/pluginmachinery/io/iface.go

```diff
@@ -41,7 +41,17 @@ type OutputReader interface {
 	Read(ctx context.Context) (*core.LiteralMap, *ExecutionError, error)
 }
 
-// All paths where various outputs produced by the task can be placed, such that the framework can directly access them.
+// Data sandbox is the actual path where the data produced by a task can be placed. It is completely optional. The advantage
+// of using this path is to provide exactly once semantics. It is guaranteed that this path is unique for every new execution
+// of a task (across retries etc) and is constant for a specific execution.
+// As of 02/20/2020 Flytekit generates this path randomly for S3. This structure proposes migration of this logic to
+// FlytePluginMachinery so that it can be used more universally outside of Flytekit.
+type OutputDataSandbox interface {
+	// This is prefix (blob store prefix or directory) where all data produced can be stored.
+	GetOutputDataSandboxPath() storage.DataReference
```

Comment:
is sandbox the best name?  That implies something to do with the sandbox deployment i feel like.

Reply:
hmm true, it is really a sandbox. Do you have any suggestions?

Reply:
I like the name sandbox for this, but lets really think of a good alternative, then i can change. Otherwise lets stick with this. So i called it DataOutputSandbox

Reply:
AtRestStorage, UserStorageLocation, FlyteHardDrive, FlyteBlackBox, FlyteRecorder, UserOutputLocation, UserOutputStorage, FlyteDurableStore, etc.  I don't really care that much, but i find some Flyte things already unnecessarily confusing and would just rather err on the side of less confusing.